core/org/commcare/core/interfaces/AbstractUserSandbox.java

```diff
@@ -0,0 +1,40 @@
+package org.commcare.core.interfaces;
+
+import org.commcare.cases.ledger.Ledger;
+import org.commcare.cases.model.Case;
+import org.javarosa.core.model.User;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+
+/**
+ *  Interface to be implemented by sandboxes for a user's CommCare instance data
+ *  @author wpride1
+ */
+
+public abstract class AbstractUserSandbox {
+
+    String syncToken;
+
+    public abstract IStorageUtilityIndexed<Case> getCaseStorage();
+
+    public abstract IStorageUtilityIndexed<Ledger> getLedgerStorage();
+
+    public abstract IStorageUtilityIndexed<User> getUserStorage();
+
+    public abstract IStorageUtilityIndexed<FormInstance> getUserFixtureStorage();
+
+    public abstract IStorageUtilityIndexed<FormInstance> getAppFixtureStorage();
```

Comment:
Would be awesome to have comments detailing what is expected to be stored in storage objects returned by these methods. Like what is the difference between user and app fixture storage?